pkg/controller/certificaterequest/certificaterequest_controller.go

```diff
@@ -111,6 +128,26 @@ func (r *CertificateRequestReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 	}()
 
+	// Explicitly fail if the certificate request has been denied
+	if cmutil.CertificateRequestIsDenied(&certificateRequest) {
+		msg := "certificate request has been denied, not signing"
+		log.Info(msg, "cr", req.NamespacedName)
+		setReadyCondition(cmmeta.ConditionFalse, cmapi.CertificateRequestReasonDenied, msg)
+		r.Recorder.Event(&certificateRequest, eventTypeWarning, cmapi.CertificateRequestReasonDenied, msg)
```

Comment:
I think we shouldn't fire an event if the CR is denied, as the approver should fire an event anyway. Saves a bit of pressure on the API.